mlflow/utils/gorilla.py

```diff
@@ -353,35 +346,31 @@ def revert(patch):
     with modifictions for autologging disablement purposes.
     """
     # If an curr_active_patch has not been set on destination class for the current patch,
-    # then there is no reverting to do.
+    # then there is no patching exist.
     curr_active_patch = _ACTIVE_PATCH % (patch.name,)
-    if not hasattr(patch.destination, curr_active_patch):
-        return
+    if curr_active_patch not in patch.destination.__dict__:
+        raise RuntimeError(f"The patch does not exist or has been reverted.")
 
-    try:
-        original = get_original_attribute(patch.destination, patch.name)
-    except AttributeError:
+    original_name = _ORIGINAL_NAME % (patch.name,)
+    if original_name not in patch.destination.__dict__:
         raise RuntimeError(
             "Cannot revert the attribute named '%s' since the setting "
             "'store_hit' was not set to True when applying the patch."
             % (patch.destination.__name__,)
         )
 
-    original_name = _ORIGINAL_NAME % (patch.name,)
-    if not getattr(patch.destination, original_name, None):
-        return
-
-    setattr(patch.destination, patch.name, original)
-
-    # We are only deleting the attribute if it is defined on the
-    # destination class as opposed to being inherited from parent class.
-    if original_name in patch.destination.__dict__:
-        delattr(patch.destination, original_name)
-    # If an curr_active_patch has been set on the destination class,
-    # then remove that attribute as well to properly clean up patched code.
-    # This is undoing the custom changes to gorilla.py's code in the `apply()`.
-    if curr_active_patch in patch.destination.__dict__:
-        delattr(patch.destination, curr_active_patch)
+    # get original attribute bypassing descriptor protocal
+    original = object.__getattribute__(patch.destination, original_name)
+
+    if patch.is_inplace_patch:
+        # restore original method
+        setattr(patch.destination, patch.name, original)
+    else:
+        # delete patched method
+        delattr(patch.destination, patch.name)
```

Comment:
Update revert logic to avoid side effect after reverting patch.

Suppose the case:
```
class A:
    def predict(self):
        pass

class B(A):
    pass
```

When we patch `B.predict`, `B.__dict__` was appended a new attribute `predict`, but after reverting the patch, the new attribute `predict` still exists in `B.__dict__`, this is a risky side effect.

Reply:
Can we create a test to verify this?